frontend/src/components/Room.jsx

```
@@ -0,0 +1,93 @@
+import { SquarePen, X } from 'lucide-react';
+import { useState } from 'react';
+
+export const Room = ({ room, setAllRooms }) => {
+  const [isEditing, setEditing] = useState(false);
+  const [newRoom, setNewRoom] = useState(room.name);
+
+  const handleEditRoom = async (e) => {
+    e.preventDefault();
+
+    if (!newRoom.trim()) {
+      setEditing(false);
+      handleDeleteRoom();
+      return;
+    }
+
+    try {
+      const response = await fetch(`http://localhost:3005/rooms/${room.id}`, {
+        method: 'PATCH',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          name: newRoom,
+          user: room.user,
+        }),
+      });
+
+      const updatedRoom = await response.json();
```

You call `await response.json()` without checking `response.ok` or status. If the server responds with an error status, you may still try to treat it as a successful update. Check `if (!response.ok)` and handle errors (show message / throw) before parsing the body to avoid silent failures or inconsistent UI updates.